catalogue_pipeline/reindexer/src/main/scala/uk/ac/wellcome/platform/reindexer/services/ReindexTargetService.scala

```diff
@@ -12,91 +12,113 @@ import com.gu.scanamo.syntax._
 
 import scala.concurrent.Future
 
-class ReindexTargetService[T <: Reindexable[String]] @Inject()(
+class ReindexTargetService @Inject()(
   dynamoDBClient: AmazonDynamoDB,
   metricsSender: MetricsSender,
-  @Flag("reindex.target.tableName") targetTableName: String,
-  @Flag("reindex.target.reindexShard") targetReindexShard: String = "default")
+  versionedDao: VersionedDao,
+  @Flag("reindex.sourceData.tableName") targetTableName: String)
     extends Logging {
 
-  type ScanamoQueryResult = Either[DynamoReadError, T]
-
-  type ScanamoQueryResultFunction =
-    (List[ScanamoQueryResult]) => Boolean
+  implicit val versionUpdater = new VersionUpdater[HybridRecord] {
+    override def updateVersion(record: HybridRecord,
+                               newVersion: Int): HybridRecord =
+      record.copy(version = newVersion)
+  }
 
-  private val gsiName = "ReindexTracker"
+  type ScanamoQueryResult = Either[DynamoReadError, HybridRecord]
+  type ScanamoQueryResultFunction = (List[ScanamoQueryResult]) => Boolean
 
-  private def scanamoUpdate(k: UniqueKey[_],
-                            updateExpression: UpdateExpression)(
-    implicit evidence: DynamoFormat[T]): Either[DynamoReadError, T] =
-    Scanamo.update[T](dynamoDBClient)(targetTableName)(k, updateExpression)
+  private val gsiName = "reindexTracker"
 
-  private def scanamoQueryStreamFunction(queryRequest: ScanamoQueryRequest,
-                                         f: ScanamoQueryResultFunction)(
-    implicit evidence: DynamoFormat[T]): ScanamoOps[List[Boolean]] =
-    ScanamoQueryStream.run[T, Boolean](queryRequest, f)
+  private def scanamoQueryStreamFunction(
+    queryRequest: ScanamoQueryRequest,
+    resultFunction: ScanamoQueryResultFunction
+  )(implicit evidence: DynamoFormat[HybridRecord]): ScanamoOps[List[Boolean]] =
+    ScanamoQueryStream.run[HybridRecord, Boolean](queryRequest, resultFunction)
 
-  private def updateVersion(requestedVersion: Int)(
+  private def updateVersion(desiredVersion: Int)(
     resultGroup: List[ScanamoQueryResult])(
-    implicit evidence: DynamoFormat[T]): Boolean = {
+    implicit evidence: VersionedDynamoFormatWrapper[HybridRecord],
+    versionUpdater: VersionUpdater[HybridRecord]): Boolean = {
     val updatedResults = resultGroup.map {
       case Left(e) => Left(e)
-      case Right(miroTransformable) => {
-        val reindexItem = Reindexable.getReindexItem(miroTransformable)
-
-        scanamoUpdate(reindexItem.hashKey and reindexItem.rangeKey,
-                      set('ReindexVersion -> requestedVersion))
+      case Right(hybridRecord) => {
+        val existingRecord =
+          versionedDao.getRecord[HybridRecord](id = hybridRecord.id)
+        existingRecord.map { possibleRecord =>
+          // getRecord() returns an Option[HybridRecord] because you may be
+          // looking up a non-existent ID; since the ID came from the table we
+          // can assume the record exists!
+          val record = possibleRecord.get
```

Comment:
Ideally we might wrap this in something that logs a clear exception in the case of failure here.